src/sentry/grouptype/grouptype.py

```diff
@@ -0,0 +1,110 @@
+from dataclasses import dataclass
+
+from sentry.event_manager import DEFAULT_GROUPHASH_IGNORE_LIMIT
+from sentry.types.issues import GroupCategory
+
+_group_type_registry = {}
+
+
+@dataclass(frozen=True)
+class GroupType:
+    type_id: int
+    slug: str
+    description: str
+    category: int
+    ignore_limit: int = DEFAULT_GROUPHASH_IGNORE_LIMIT
+
+    def __init_subclass__(cls, **kwargs):
+        super().__init_subclass__(**kwargs)
+        _group_type_registry[cls.__name__] = cls
```

Comment:
I think that we should probably have the key here be the `type_id`, since we never want to duplicate those.

We should probably also check that a given `type_id` hasn't already been registered and throw an exception if it has.